lib/hammer_cli_foreman/hosts/common_update_options.rb

```diff
@@ -100,7 +100,13 @@ def request_params
         end
 
         if options["option_image_id"]
-          compute_resource_id = params['host']['compute_resource_id'] || ::HammerCLIForeman::ComputeResources.get_host_compute_resource_id(params['id'])
+          if params['host']['compute_resource_id']
+            compute_resource_id = params['host']['compute_resource_id']
+          elsif  params['id']
+            compute_resource_id = ::HammerCLIForeman::ComputeResources.get_host_compute_resource_id(params['id'])
+          elsif params['host']['hostgroup_id']
+            compute_resource_id = ::HammerCLIForeman::ComputeResources.get_hostgroup_compute_resource_id(params['host']['hostgroup_id'])
+          end
```

Comment:
Nit: Sorry for adding this nit, it is a personal choice and feel free to ignore :)
When we have multiple if/else/ifelse, we can use the `case` statement. Something like:

```
if options['option_image_id']
  compute_resource_id = case [params['host']['compute_resource_id'], ::HammerCLIForeman::ComputeResources.get_host_compute_resource_id(params['id'])]
                          when params['host']['compute_resource_id'] then params['host']['compute_resource_id']
                          when params['id'] then ::HammerCLIForeman::ComputeResources.get_host_compute_resource_id(params['id'])
                          when params['host']['hostgroup_id'] then ::HammerCLIForeman::ComputeResources.get_hostgroup_compute_resource_id(params['host']['hostgroup_id'])
                        end
  raise ArgumentError, "Missing argument for 'compute_resource'" if compute_resource_id.nil?
  image_uuid =  ::HammerCLIForeman::ComputeResources.get_image_uuid(compute_resource_id, options["option_image_id"])
  params['host']['compute_attributes']['image_id'] = image_uuid
end

I haven't tested this, but it looks like it. @shiramax @kgaikwad let me know your thoughts on this!

Reply:
thanks @rahulbajaj0509 , Done :)

Reply:
@rahulbajaj0509, @shiramax 

In my opinion, +1 to if/else/ifelse loop that was there previously.
It was more readable.